gobblin-yarn/src/main/java/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -102,6 +118,19 @@
 
   private static final Splitter SPLITTER = Splitter.on(',').omitEmptyStrings().trimResults();
 
+  // The set of Yarn application types this class is interested in. This is used to
+  // lookup the application this class has launched previously upon restarting.
+  private static final Set<String> APPLICATION_TYPES = ImmutableSet.of("YARN");
```

Comment:
Can we use a custom application type instead? That gives us a more precise way of finding the correct running application. You can use https://hadoop.apache.org/docs/r2.6.1/api/org/apache/hadoop/yarn/api/records/ApplicationSubmissionContext.html#setApplicationType(java.lang.String) to set the application type.